### PR TITLE
[wip] just another private cluster

### PR DESCRIPTION
**Pull request by j0npulsifer**

### CODE REVIEW

**Comment by j0npulsifer** [comment]
Ran Plan for dir: `gcp/projects/kubesec` workspace: `default`

**Plan Error**
<details><summary>Show Output</summary>

```
running "/usr/local/bin/terraform init -input=false -no-color -upgrade" in "/home/atlantis/.atlantis/repos/jonpulsifer/cloudlab/20/default/gcp/projects/kubesec": exit status 1
Upgrading modules...
- gke-controlplane in 
Downloading github.com/jonpulsifer/terraform-modules for lab...
- lab in .terraform/modules/lab/gce-igm-shielded
Downloading github.com/jonpulsifer/terraform-modules for network...
- network in .terraform/modules/network/gce-vpc
Downloading github.com/jonpulsifer/terraform-modules for ubuntu-nodes...
- ubuntu-nodes in .terraform/modules/ubuntu-nodes/gke-nodepool

Error: Unreadable module directory

Unable to evaluate directory symlink: lstat ../../../../terraform-modules: no
such file or directory


Error: Failed to read module directory

Module directory  does not exist or cannot be read.


Error: Unreadable module directory

Unable to evaluate directory symlink: lstat ../../../../terraform-modules: no
such file or directory


Error: Failed to read module directory

Module directory  does not exist or cannot be read.


Error: Unreadable module directory

Unable to evaluate directory symlink: lstat ../../../../terraform-modules: no
such file or directory


Error: Failed to read module directory

Module directory  does not exist or cannot be read.


Error: Unreadable module directory

Unable to evaluate directory symlink: lstat ../../../../terraform-modules: no
such file or directory


Error: Failed to read module directory

Module directory  does not exist or cannot be read.


Error: Unreadable module directory

Unable to evaluate directory symlink: lstat ../../../../terraform-modules: no
such file or directory


Error: Failed to read module directory

Module directory  does not exist or cannot be read.


```
</details>

**Comment by j0npulsifer** [comment]
Ran Plan for dir: `gcp/projects/kubesec` workspace: `default`

**Plan Error**
<details><summary>Show Output</summary>

```
running "/usr/local/bin/terraform plan -input=false -refresh -no-color -out \"/home/atlantis/.atlantis/repos/jonpulsifer/cloudlab/20/default/gcp/projects/kubesec/default.tfplan\"" in "/home/atlantis/.atlantis/repos/jonpulsifer/cloudlab/20/default/gcp/projects/kubesec": exit status 1
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

google_storage_bucket.cloud-lab: Refreshing state... [id=cloud-lab]
google_project_iam_member.cloudbuild: Refreshing state... [id=kubesec/roles/container.admin/serviceAccount:821879192255@cloudbuild.gserviceaccount.com]
google_dns_managed_zone.pulsifer-dev: Refreshing state... [id=projects/kubesec/managedZones/pulsifer-dev]
module.gke-controlplane.data.google_project.current: Refreshing state...
module.network.google_compute_network.network: Refreshing state... [id=projects/kubesec/global/networks/lab]
module.lab.google_service_account.igm: Refreshing state... [id=projects/kubesec/serviceAccounts/lab-igm@kubesec.iam.gserviceaccount.com]
google_project_organization_policy.bool-policies["iam.disableServiceAccountKeyCreation"]: Refreshing state... [id=kubesec:constraints/iam.disableServiceAccountKeyCreation]
data.google_container_engine_versions.cloudlab: Refreshing state...
data.google_iam_policy.gcs-cloud-lab: Refreshing state...
module.lab.data.google_compute_image.trusted-image: Refreshing state...
google_monitoring_alert_policy.stackdriver-free-storage-warning: Refreshing state... [id=projects/kubesec/alertPolicies/5586130003877248095]
google_project_organization_policy.list-policies-allow["iam.allowedPolicyMemberDomains"]: Refreshing state... [id=kubesec:constraints/iam.allowedPolicyMemberDomains]
google_service_account.cloudlab: Refreshing state... [id=projects/kubesec/serviceAccounts/cloudlab@kubesec.iam.gserviceaccount.com]
module.lab.data.google_project.current: Refreshing state...
google_dns_managed_zone.k8s-pulsifer-dev: Refreshing state... [id=k8s-pulsifer-dev]
google_monitoring_alert_policy.stackdriver-free-storage-alert: Refreshing state... [id=projects/kubesec/alertPolicies/6798117528715859814]
google_project_organization_policy.bool-policies["compute.disableGuestAttributesAccess"]: Refreshing state... [id=kubesec:constraints/compute.disableGuestAttributesAccess]
google_project_organization_policy.bool-policies["iam.disableServiceAccountCreation"]: Refreshing state... [id=kubesec:constraints/iam.disableServiceAccountCreation]
google_project_organization_policy.list-policies-allow["serviceuser.services"]: Refreshing state... [id=kubesec:constraints/serviceuser.services]
google_project_organization_policy.list-policies-allow["compute.vmExternalIpAccess"]: Refreshing state... [id=kubesec:constraints/compute.vmExternalIpAccess]
google_project_organization_policy.list-policies-values["compute.trustedImageProjects"]: Refreshing state... [id=kubesec:constraints/compute.trustedImageProjects]
google_project_organization_policy.list-policies-values["gcp.resourceLocations"]: Refreshing state... [id=kubesec:constraints/gcp.resourceLocations]

Error: Error reading Storage Bucket "cloud-lab": googleapi: Error 403: seawall-ae-1234567@appspot.gserviceaccount.com does not have storage.buckets.get access to cloud-lab., forbidden



Error: Error reading Service Account "projects/kubesec/serviceAccounts/cloudlab@kubesec.iam.gserviceaccount.com": googleapi: Error 403: Permission iam.serviceAccounts.get is required to perform this operation on service account projects/kubesec/serviceAccounts/cloudlab@kubesec.iam.gserviceaccount.com., forbidden



Error: Error reading DNSManagedZone "k8s-pulsifer-dev": googleapi: Error 403: Forbidden, forbidden



Error: Error reading Organization policy for projects/kubesec: googleapi: Error 403: The caller does not have permission, forbidden



Error: Error reading Organization policy for projects/kubesec: googleapi: Error 403: The caller does not have permission, forbidden



Error: Error reading MonitoringAlertPolicy "projects/kubesec/alertPolicies/5586130003877248095": googleapi: Error 403: Permission denied (or the resource may not exist).



Error: Error reading ComputeNetwork "projects/kubesec/global/networks/lab": googleapi: Error 403: Required 'compute.networks.get' permission for 'projects/kubesec/global/networks/lab', forbidden



Error: Error reading Service Account "projects/kubesec/serviceAccounts/lab-igm@kubesec.iam.gserviceaccount.com": googleapi: Error 403: Permission iam.serviceAccounts.get is required to perform this operation on service account projects/kubesec/serviceAccounts/lab-igm@kubesec.iam.gserviceaccount.com., forbidden



Error: Error reading Organization policy for projects/kubesec: googleapi: Error 403: The caller does not have permission, forbidden



Error: Error reading Organization policy for projects/kubesec: googleapi: Error 403: The caller does not have permission, forbidden



Error: Error reading Organization policy for projects/kubesec: googleapi: Error 403: The caller does not have permission, forbidden



Error: Error reading Organization policy for projects/kubesec: googleapi: Error 403: The caller does not have permission, forbidden



Error: Error reading Organization policy for projects/kubesec: googleapi: Error 403: The caller does not have permission, forbidden



Error: Error reading Organization policy for projects/kubesec: googleapi: Error 403: The caller does not have permission, forbidden



Error: Error reading DNSManagedZone "projects/kubesec/managedZones/pulsifer-dev": googleapi: Error 403: Forbidden, forbidden



Error: Error reading MonitoringAlertPolicy "projects/kubesec/alertPolicies/6798117528715859814": googleapi: Error 403: Permission denied (or the resource may not exist).



```
</details>

File: gcp/projects/kubesec/policy.tf [hunk 1]
```diff
@@ -3,6 +3,7 @@ resource "google_project_organization_policy" "bool-policies" {
     "iam.disableServiceAccountCreation" : false,
     "iam.disableServiceAccountKeyCreation" : false,
     "compute.disableGuestAttributesAccess" : false,
+    "compute.requireShieldedVm" : false,
```
Comment (j0npulsifer) [inline gcp/projects/kubesec/policy.tf:6]:
```suggestion
```

the default pool deploys without a shielded vm so this is required

File: gcp/projects/kubesec/policy.tf [hunk 2]
```diff
@@ -35,8 +36,7 @@ resource "google_project_organization_policy" "list-policies-values" {
       "projects/gce-uefi-images",
     ],
     "gcp.resourceLocations" : [
-      "us-east4-b",
-      "us-east4"
+      "in:us-east4-locations",
```
Comment (jonpulsifer) [inline gcp/projects/kubesec/policy.tf:39]: regional node pool means we need all the zones

**Pull request merged**